added VSCode & Docker integration

brakmic opened the pull request:
Fixes https://github.com/microsoft/BosqueLanguage/issues/337

Changes:
This PR adds VSCode & Docker integration option for interested developers. This way they wouldn't have to manually clone, compile and install their development environments. Future updates could be integrated more easily as Docker would take care of these tasks. 

There is also a small readme in a new subdir `docs/extras/docker.md`.

mrkmarron commented:
Overall looks good. 

One small tweak is to add the git lfs setup: so the clone will grab any binaries we have (right now just Z3):
```
git lfs install
```

Once that is done I think we can merge.

brakmic commented:
@mrkmarron 

Git LFS is now getting installed. Have also cleaned up Dockerfile a bit and added a new build argument, **NODE_VERSION**. Also, docker.md got updated a bit.
